src/rgw/rgw_op.cc

```diff
@@ -7958,6 +7958,11 @@ int RGWHandler::do_init_permissions(const DoutPrefixProvider *dpp, optional_yiel
 
 int RGWHandler::do_read_permissions(RGWOp *op, bool only_bucket, optional_yield y)
 {
+  if (op->get_type() != RGW_OP_CREATE_BUCKET &&
+      rgw::sal::Bucket::empty(s->bucket)) {
+    return -ERR_NO_SUCH_BUCKET;
```

Comment:
all of the verify jobs in https://pulpito.ceph.com/cbodley-2022-10-18_16:36:54-rgw-wip-cbodley-testing-distro-default-smithi/ failed during run-reshard.sh:

> 2022-10-18T17:57:56.641 INFO:tasks.workunit.client.0.smithi037.stderr:botocore.errorfactory.NoSuchBucket: An error occurred (NoSuchBucket) when calling the ListBuckets operation: Unknown

i think we need to filter out `RGW_OP_LIST_BUCKETS` here too

Reply:
Valid point, I'll rework this!

Reply:
This is helpful, I think I need to filter out all ops that use the service endpoint and don't need a bucket at all. I'll try this and update